[
    {
        "content": "<p>Hi, I have enabled the backtracing but it traces some functions, not all, is there any way to fix it to show all functions that call  each other?</p>",
        "id": 217274470,
        "sender_full_name": "hossein dindar",
        "timestamp": 1605795389
    },
    {
        "content": "<p>are you talking about the logging (RUSTC_LOG) or about backtraces when the compiler crashes?</p>",
        "id": 217274766,
        "sender_full_name": "Joshua Nelson",
        "timestamp": 1605795544
    },
    {
        "content": "<p>no about crashing, when a program has error and the compiler panics and shows stack backtrace</p>",
        "id": 217274985,
        "sender_full_name": "hossein dindar",
        "timestamp": 1605795636
    },
    {
        "content": "<p>hmm, maybe <code>RUST_BACKTRACE=full</code>?</p>",
        "id": 217275337,
        "sender_full_name": "Joshua Nelson",
        "timestamp": 1605795786
    },
    {
        "content": "<p>no it didn't help, actually it shows that <code>parse_block_tail </code> has called f.e, but this function calls <code>parse_full_stmt</code>, but it doesn't show in tracing</p>",
        "id": 217277099,
        "sender_full_name": "hossein dindar",
        "timestamp": 1605796502
    },
    {
        "content": "<p>that sounds like you're missing debug info for the compiler</p>",
        "id": 217277426,
        "sender_full_name": "Joshua Nelson",
        "timestamp": 1605796653
    },
    {
        "content": "<p>try <code>debug = true</code> in config.toml</p>",
        "id": 217277451,
        "sender_full_name": "Joshua Nelson",
        "timestamp": 1605796665
    },
    {
        "content": "<p>yeah it was false, but do I need to clear the build and then build again?</p>",
        "id": 217277685,
        "sender_full_name": "hossein dindar",
        "timestamp": 1605796774
    },
    {
        "content": "<p>unfortunately it will rebuild, yes. You don't have to run <code>x.py clean</code> though, x.py knows that debug = true invalidates the cache</p>",
        "id": 217277731,
        "sender_full_name": "Joshua Nelson",
        "timestamp": 1605796802
    },
    {
        "content": "<p>hmm but I did <code>./x.py build</code> now and the result is same</p>",
        "id": 217277838,
        "sender_full_name": "hossein dindar",
        "timestamp": 1605796842
    },
    {
        "content": "<p>oh sorry no, wait</p>",
        "id": 217277873,
        "sender_full_name": "hossein dindar",
        "timestamp": 1605796856
    },
    {
        "content": "<p>yeah the result is same, the command I run is this: <code>RUST_BACKTRACE=full RUSTC_LOG=\"rustc_parse=debug\" rustc +stage1 ./src/test/ui/parser/mistyped_reserved_token.rs -Z treat-err-as-bug=1</code></p>",
        "id": 217278116,
        "sender_full_name": "hossein dindar",
        "timestamp": 1605796968
    },
    {
        "content": "<p>and what's the output you're seeing?</p>",
        "id": 217278592,
        "sender_full_name": "Joshua Nelson",
        "timestamp": 1605797173
    },
    {
        "content": "<p>the output is same with previous result</p>",
        "id": 217278659,
        "sender_full_name": "hossein dindar",
        "timestamp": 1605797203
    },
    {
        "content": "<p>f.e I don't see that <code>parse_full_stmt</code></p>",
        "id": 217278721,
        "sender_full_name": "hossein dindar",
        "timestamp": 1605797225
    }
]